File: src/ProjectR/content/scripts/spells/GhostNova.lua
function GetTargetType()
    return Enemies
end

function GetName()
  return "Ghost Nova"
end

function GetDescription()
  return "Unleash several ghosts to haunt your enemies.\nChance to instantly kill targets.\nHeals you for 1/4th the damage done."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .96
end

function GetDelay()
  return .1
end

function GetSpellType()
  return Magical
end

function GetMasteries()
  return { DRK }
end

function SpellEffect()
  d:ApplyDebuff(DTH, 66)

  local damage = (4 * aMD * (aDRK/100) - dMR) * (100/dDRK)  
  a:Heal(damage / 4)
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/GhostNova.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class GhostNova : SpellScriptBase
    {
        public override string Name { get { return "Ghost Nova"; } }
        public override string Description { get { return "Unleash several ghosts to haunt your enemies.\nChance to instantly kill targets.\nHeals you for 1/4th the damage done."; } }

        public override TargetType TargetType { get { return TargetType.Enemies; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { EleMastery.DRK, }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 96; } }
        public override double Delay { get { return .1; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = (4 * cMD * (cDRK / 100) - tMR) * (100 / tDRK);
            Heal(caster, damage / 4);
            DealDamage(damage);
            TryToApplyDebuff(DebuffResistance.DTH, 66);
        }
    }
}